src/async/prepare.js

```diff
@@ -15,6 +15,7 @@ import {
 } from 'react-is';
 
 import isReactCompositeComponent from './utils/isReactCompositeComponent';
+import isReactFunctionalComponent from './utils/isReactFunctionalComponent';
```

Comment:
This is a new case, correct? Would be amazing if we could have a unit test for this.

Reply:
Added tests for both utils